include/nfc/tnep/tag.rst

```diff
@@ -0,0 +1,141 @@
+
+TNEP for tag device
+###################
+
+The tag library provides functionality to implement the Tag NDEF Exchange Protocol (TNEP) for an NFC tag.
+
+The tag library is used in the :ref:`nfc_tnep_tag` sample.
+
+Initialization
+**************
+
+Before using the tag library in a tag application, you must configure the NDEF message buffers and define the TNEP services.
+
+Configuring NDEF message buffers
+================================
+
+The tag library communicates with the tag using NDEF message buffers.
+The buffers are initialized all at the same time by :cpp:func:`nfc_tnep_tag_tx_msg_buffer_register`.
+
+To use the NDEF message buffers, you must configure the following elements:
+
+* Provide a pointer to two TX buffers - one for emulating the NDEF message and one for encoding the new NDEF message.
+* Set a maximum buffer size in bytes, which is valid for both buffers.
+
+The buffers must be linked with the tag layer in the application.
+
+When the polling device writes a new NDEF message to the tag buffer, the application must inform the tag library by calling :cpp:func:`nfc_tnep_tag_rx_msg_indicate`.
+
+Defining TNEP services
+======================
+
+TNEP services are capabilities and features provided to the adjacent upper layer.
+They must be defined for the tag library to create the initial NDEF message.
+At least one TNEP service must be defined for the message to be created.
+
+To create a TNEP service, use the :c:macro:`NFC_TNEP_TAG_SERVICE_DEF` macro.
+
+After defining the services, create a table that lists all of them.
+You can use :c:macro:`NFC_TNEP_TAG_SERVICE` to get the names of the created services.
+
+The service table is given to the :cpp:func:`nfc_tnep_tag_initial_msg_create` function and the Service Parameter record of each service is inserted into the initial TNEP message.
+A Service Parameter record contains parameters for communicating with the service.
+
+See the following code for an example of service definition and TNEP initialization:
+
+.. code-block:: c
+
+	#define NFC_TNEP_TAG_SERVICE_DEF(service_name, uri, uri_length, mode, t_wait, n_wait, mac_msg_size
+								 select_cb, deselect_cb message_cb, timeout_cb, error_cb)
+
+	NFC_NDEF_MSG_DEF(ndef_msg, 16);
+
+	static struct nfc_tnep_tag_service services[] = {
+		NFC_TNEP_TAG_SERVICE(service_name),
+	};
+
+	nfc_tnep_tx_msg_buffer_register(buffer, swap_buffer, buffer_length);
+
+	nfc_tnep_init(event, event_cnt, &NFC_NDEF_MSG(ndef_msg),
+		      nfc_txt_rw_payload_set);
+
+	nfc_tXt_setup();
+
+	nfc_tnep_tag_initial_msg_create(services, ARRAY_SIZE(services), NULL, 0);
+
+	nfc_tXt_emulation_start();
+
+Configuration
+*************
+After initialization, the tag library is in the Service Ready state and the NDEF message buffer contains Service Parameter records provided with the initial TNEP message.
+
+The initial TNEP message can also include optional NDEF records that can interact with the polling device that does not support the protocol.
```

Comment:
Not sure I understand. Is this in case you are using a polling device that does not support TNEP?

Reply:
Yes, the Poller will simply skip records that it does not understand (in this case TNEP records).